sdk/python/lib/pulumi/automation/_stack_settings.py

```diff
@@ -28,7 +28,7 @@ class StackSettings:
     secrets_provider: Optional[str]
     encrypted_key: Optional[str]
     encryption_salt: Optional[str]
-    config: Optional[Mapping[str, Any]]
+    config: Optional[Dict[str, Union[str, StackSettingsSecureConfigValue]]]
```

Comment:
Shouldn't this technically be `Any` though since we can have nested objects in the config?

Reply:
For now, in automation API, we don't unmarshal objects and arrays from config, and they are read in as strings. There's an [open issue](https://github.com/pulumi/pulumi/issues/5506) tracking this. So, this matches the current reality and the other SDKs.

Reply:
Ah, right. Thanks for clarifying.

Reply:
Actually, you're right, kinda. I was thinking of `pulumi config` commands in my earlier answer. 

When the settings are parsed from json/yaml, in python the nested objects will parse to dicts and lists. So we should be recursing through if it's an object or a list to find the "secure" and regular config values. That means it's not just the type that's wrong, but also the serde for nested stack settings config in general 😞 

Gonna have to look at this with fresh eyes in the morning

Reply:
Okay, I updated the types. The deserialization is still not perfect but the issue falls under the umbrella covered in #5506.